src/win/win_lang.c

```diff
@@ -76,6 +78,81 @@ progsett_fill_languages(HWND hdlg)
 	SendMessage(lang_combo, CB_SETCURSEL, enum_helper, 0);
 }
 
+/* Load available iconsets */
+static void
+progsett_fill_iconsets(HWND hdlg)
+{
+	HWND icon_combo = GetDlgItem(hdlg, IDC_COMBO_ICON); 
+	
+	/* Add the default one */
+	wchar_t buffer[512] = L"(";
+	wcscat(buffer, plat_get_string(IDS_2090));
+	wcscat(buffer, L")");
+	
+	SendMessage(icon_combo, CB_RESETCONTENT, 0, 0);
+	SendMessage(icon_combo, CB_ADDSTRING, 0, (LPARAM)buffer);
+	SendMessage(icon_combo, CB_SETITEMDATA, 0, (LPARAM)strdup(""));
+	
+	int combo_index = -1;
+	
+	/* Find for extra ones */
+	HANDLE hFind;
+	WIN32_FIND_DATA data;
+	
+	char icon_path_root[512];
+	plat_get_icons_path(icon_path_root);
+	
+	wchar_t search[512];
+	mbstowcs(search, icon_path_root, strlen(icon_path_root) + 1);
+	wcscat(search, L"*.*");
+	
+	hFind = FindFirstFile((LPCWSTR)search, &data);
+	
+	if (hFind != INVALID_HANDLE_VALUE) {
+		do {
+			if (wcscmp(data.cFileName, L".") && wcscmp(data.cFileName, L"..") && 
+			  (data.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY))
+			{
+				wchar_t temp[512] = {0}, dispname[512] = {0};
+				mbstowcs(temp, icon_path_root, strlen(icon_path_root) + 1);
+				wcscat(temp, data.cFileName);
+				wcscat(temp, L"\\iconinfo.txt");
+								
+				wcscpy(dispname, data.cFileName);
+				FILE *fp = _wfopen(temp, L"r");
+				if (fp)
+				{
+					char line[512] = {0};
+					if (fgets(line, 511, fp))
+					{
+						mbstowcs(dispname, line, strlen(line) + 1);
+					}
+					
+					fclose(fp);
+				}
+				
+				char filename[512];
+				wcstombs(filename, data.cFileName, 511);
+				
+				int index = SendMessage(icon_combo, CB_ADDSTRING, 0, (LPARAM)dispname);
+				SendMessage(icon_combo, CB_SETITEMDATA, index, (LPARAM)(strdup(filename)));
+				
+				if (!strcmp(filename, icon_set))
+					combo_index = index;
+			}
+		} while (FindNextFile(hFind, &data));
+		FindClose(hFind);
+	}
+	
+	if (combo_index == -1)
+	{
+		combo_index = 0;
+		strcpy(temp_icon_set, "");
+	}
+	
+	SendMessage(icon_combo, CB_SETCURSEL, combo_index, 0);
+}
+
```

Comment:
Seems a bit strange to put this in the Language dialog; if your intent is to turn it into a global settings dialog then perhaps the file should be renamed.